### PR TITLE
Faqs

### PR DESCRIPTION
**Pull request by kartikeya2609**
Faqs

Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Added few faqs

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [X] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by ravilach** [comment]
@SushrutHarness / @Debanitrkl  / @nofarblue mind taking a look at and approving? Since this touches multiple modules, a few folks got tagged in. Thanks!

**Review by brian-f-harness — APPROVED** [review]
approve for PL

**Review by Debanitrkl — APPROVED** [review]
LGTM for IDP

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 1]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+### Can user import test results from a generic 'mvn clean deploy' Run step into the test Dashboard? Or must use the Unit Test step to display them on that tab?
+Yes, you can use the Run step to run your maven command, and upload the test results. There is a field for this in the run step (like in the run tests step). Please see
+https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/containerized-steps/run-step/#report-paths for more information.
+
+The RunTests allow you to not only run your tests, but also to use Test Intelligence to run only unit tests that are relevant to the code changes made, thus cutting down the test cycle time.
+ 
+### how can user add the exclusion for jacoco code coverage report?
+pom.xml allows using environment variable references with syntax like below:
+ 
+`${env. VARIABLE_NAME} `
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2284]:
```suggestion
`${env.VARIABLE_NAME}`
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 2]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+when:
+  stageStatus: Success
+  condition: <+strategy.iteration> == 0
+
+```
+This ensures the Save Cache step runs only if it's the first instance (iteration 0) of the parallel stages and if the current stage execution is successful.
+
+
+### How does caching work when cloning pipelines with Save/Restore Cache steps?
+When user clone a pipeline with caching steps, the cache keys generated by the cloned pipeline use the original pipeline's cache key as a prefix. For instance, if the original cache key is some-cache-key, the cloned pipeline might use some-cache-key2. This prevents accidental interference between cache keys of the original and cloned pipelines.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2264]:
```suggestion
When the user clones a pipeline with caching steps, the cache keys generated by the cloned pipeline use the original pipeline's cache key as a prefix. For instance, if the original cache key is `some-cache-key`, the cloned pipeline might use `some-cache-key2`. This prevents accidental interference between cache keys of the original and cloned pipelines.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 3]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
 ### What actions can I take to address the extended runtime of a Docker command, particularly when encountering timeouts during microdnf updates?
 The prolonged runtime of Docker commands, often exacerbated by timeouts during microdnf updates, can be primarily attributed to network issues. To address this, you should verify the connectivity of the repomirror being utilized. Additionally, you could examine and potentially adjust the timeouts configured on the repomirror to enhance performance. These steps can help mitigate the impact of network-related delays and improve the efficiency of Docker operations.
 
+### How can user access the secrets as files in a Docker build without writing them to layers?
+In the ‘build and push’ steps used to build Docker images have a context field in them. User can use the context field in the build and push steps, which mounts the current directory at /harness. By copying your files to a specific directory and then mounting it, you can avoid writing secrets into the Docker image layers.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2247]:
```suggestion
The **build and push** steps used to build Docker images have a context field. Users can use the context field in the build and push steps to mount the current directory at `/harness`. By copying your files to a specific directory and then mounting them, you can avoid writing secrets into the Docker image layers.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 4]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+### How can user prevent conflicts with Save Cache steps in parallel stages?
+Skip the Save Cache step in all parallel stages except one.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2250]:
```suggestion
Skip the **Save Cache** step in all parallel stages except one.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 5]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+Use conditional execution in the Save Cache step to ensure it runs only in one instance of the parallel stages.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2251]:
```suggestion
Use conditional execution in the **Save Cache** step to ensure it runs only in one instance of the parallel stages.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 6]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+### What conditional execution user add to the Save Cache step to handle parallel stages?
+```
+when:
+  stageStatus: Success
+  condition: <+strategy.iteration> == 0
+
+```
+This ensures the Save Cache step runs only if it's the first instance (iteration 0) of the parallel stages and if the current stage execution is successful.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2260]:
```suggestion
This ensures the **Save Cache** step runs only if it's the first instance (iteration 0) of the parallel stages and if the current stage execution is successful.
```

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 7]
```diff
@@ -2243,6 +2243,48 @@ curl -X POST -i -H "Authorization: Bearer AUTH_TOKEN" https://<DRONE_SERVER_FQDN
...
+
+
+### How does caching work when cloning pipelines with Save/Restore Cache steps?
+When user clone a pipeline with caching steps, the cache keys generated by the cloned pipeline use the original pipeline's cache key as a prefix. For instance, if the original cache key is some-cache-key, the cloned pipeline might use some-cache-key2. This prevents accidental interference between cache keys of the original and cloned pipelines.
+
+### How can user ensure correct cache handling when cloning pipelines?
+Enable separators (/) for GCS cache keys by setting PLUGIN_ENABLE_SEPARATOR: true in your pipeline's stage variables.
```
Comment (dewan-ahmed) [inline kb/continuous-integration/continuous-integration-faqs.md:2267]: Is this question specific to **Save Cache to GCS**? If yes, the question need to reflect that.

**Review by dewan-ahmed — COMMENTED** [review]
I've added few suggestions and comments for CI section.

**Comment by SushrutHarness** [comment]
In order to get your PRs merged quicker in the future, we recommend that you split them by module. Additionally, CD FAQs require you to separate them by swim lane and add them to their respective /docs/ pages.

**Pull request closed without merging**